src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/validate/PNCIdValidator.kt

```diff
@@ -1,14 +1,25 @@
 package uk.gov.justice.digital.hmpps.personrecord.validate
 
 import org.slf4j.LoggerFactory
+import uk.gov.justice.digital.hmpps.personrecord.service.TelemetryService
+import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType
 import java.math.BigInteger
 
 const val PNC_REGEX = "\\d{4}(/?)\\d{7}[A-Z]{1}\$"
 
-class PNCIdValidator {
+class PNCIdValidator(private val telemetryService: TelemetryService) {
 
-  fun isValid(pncIdentifier: String): Boolean {
-    return isValidFormat(pncIdentifier) && hasValidCheckDigit(pncIdentifier)
+  fun isValid(pncIdentifier: String?): Boolean {
+    if (pncIdentifier.isNullOrEmpty()) {
+      telemetryService.trackEvent(TelemetryEventType.NEW_CASE_MISSING_PNC, emptyMap())
```

Comment:
This implies the case is missing the PNC and may not be correct for other services. If we want the validator should send the event then the event name should be generic or move to the calling services.

Reply:
Fair point, I didn't want to change the existing functionality. Perhaps something like `TelemetryEventType.MISSING_PNC`?
Would need to run it past a BA

Reply:
This assumes that we are going to get and validate PNCs from other sources

Reply:
I suggest that we hold off renaming the event until we have another source of PNCs - at the moment it is correct

Reply:
I have renamed the event